### DISCUSSION TITLE
Arama İçin Kullanılacak Yöntemin Değerlendirilmesi

### DISCUSSION THREAD

**Discussion by sertacates** [IFL-Elestirmenler/FilmAB#16] [Ideas]
Gençler, literatür çalışmamıza göre yöntem kural bazlı mı olacak, derin öğrenme (sentence similarity) mi olacak, https://dergipark.org.tr/tr/pub/ajit-e/issue/68716/1012354 buradaki gibi TF-IDF algoritması ve Kosinüs benzerliği mi kullanacağız hızlıca bir çalışmalarınızın durumunu paylaşabilir misiniz? Zoom toplantımızda görevleri dağıttık, durum ne aşama?

**Comment by fahriyegulolur** [comment]
Hocam ben biraz ilerledim. Github üzerinden biraz kodlara baktım, nltk kullanarak özetten bir şeyler çıkartabiliyorum. Öncelikle her bir terim için TF yani terim frekansına baktım, arama için anlamlı farklılık göstermeyecek veya aramayı olumlu etkilemeyecek edat bağlaç gibi terimlerin TF'si yüksek çıkıyor, bizim için değerli olan anahtar kelimeleri bulabilmemiz için IDF yani ters doküman frekansına da bakmamız gerekiyor. Anahtar kelimeleri çıkardıktan sonra nltk içinde linear_kernel ve cosine_similarity ye bakmak gerekiyor. https://scikit-learn.org/stable/modules/metrics.html buradan biraz ilerledim. Derin öğrenme kısmında https://huggingface.co/tasks/sentence-similarity ile benim bulduklarımı karşılaştırabiliriz. Hangisi daha iyi sonuç verirse ondan ilerleriz. Olur mu? Ali Batu, hocam, ne dersiniz?
